File: frontend/src/api/dto/deviceRoutesResponse.ts
import type { DeviceRideDto } from './deviceRideDto.ts'
import type { DeviceRouteDto } from './deviceRouteDto.ts'

/**
 * Response containing rides and routes for device applications
 */
export interface DeviceRoutesResponse {
  /** Upcoming rides with route entries */
  rides: DeviceRideDto[]
  /** Latest standalone routes */
  routes: DeviceRouteDto[]
}

File: backend/src/main/java/fr/pedalons/dto/device/response/DeviceRoutesResponse.java
package fr.pedalons.dto.device.response;

import fr.pedalons.dto.validation.ValidateSchema;
import java.util.List;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Response containing rides and routes for device applications")
@Builder
@ValidateSchema
public record DeviceRoutesResponse(
    @Schema(description = "Upcoming rides with route entries", required = true)
        List<DeviceRideDto> rides,
    @Schema(description = "Latest standalone routes", required = true)
        List<DeviceRouteDto> routes) {}
